prompt_toolkit/widgets/base.py

```diff
@@ -508,6 +509,15 @@ def _(event):
         def _(event):
             self.current_value = self.values[self._selected_index][0]
 
+        @kb.add(Keys.Any)
+        def _(event):
+            # We first check values after the selected value, then all values
+            for value in self.values[self._selected_index + 1:] + self.values:
+                if value[1].startswith(event.data):
+                    self._selected_index = self.values.index(value)
+                    return True
```

Comment:
One last remark: key bindings return None. So a simple return ``return`` here would do it, and you can remove the next line. Thanks!